frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx

```diff
@@ -1952,3 +1939,76 @@ const checkCustomEventConversionGoalHasSessionIdsHelper = async (
         setConversionGoalWarning(null)
     }
 }
+
+const setFilters = (
+    actions: webAnalyticsLogicType['actions'],
+    values: webAnalyticsLogicType['values'],
+    { productTab = ProductTab.ANALYTICS }: { productTab?: ProductTab },
+    {
+        filters,
+        'conversionGoal.actionId': conversionGoalActionId,
+        'conversionGoal.customEventName': conversionGoalCustomEventName,
+        date_from,
+        date_to,
+        interval,
+        device_tab,
+        source_tab,
+        graphs_tab,
+        path_tab,
+        geography_tab,
+        path_cleaning,
+        filter_test_accounts,
+        compare_filter,
+    }: Record<string, any>
+): void => {
+    const parsedFilters = isWebAnalyticsPropertyFilters(filters) ? filters : undefined
+
+    if (parsedFilters && !objectsEqual(parsedFilters, values.webAnalyticsFilters)) {
+        actions.setWebAnalyticsFilters(parsedFilters)
+    }
+    if (
+        conversionGoalActionId &&
+        conversionGoalActionId !== (values.conversionGoal as ActionConversionGoal)?.actionId
+    ) {
+        actions.setConversionGoal({ actionId: parseInt(conversionGoalActionId, 10) })
+    } else if (
+        conversionGoalCustomEventName &&
+        conversionGoalCustomEventName !== (values.conversionGoal as CustomEventConversionGoal)?.customEventName
+    ) {
+        actions.setConversionGoal({ customEventName: conversionGoalCustomEventName })
+    }
+    if (
+        (date_from && date_from !== values.dateFilter.dateFrom) ||
+        (date_to && date_to !== values.dateFilter.dateTo) ||
+        (interval && interval !== values.dateFilter.interval)
+    ) {
+        actions.setDatesAndInterval(date_from, date_to, interval)
+    }
+    if (device_tab && device_tab !== values._deviceTab) {
+        actions.setDeviceTab(device_tab)
+    }
```

Comment:
I would love so much if our style allowed this to be inlined, it's just so clean. One can only wonder ⭐ 

```suggestion
    if (device_tab && device_tab !== values._deviceTab) actions.setDeviceTab(device_tab)
```

Reply:
And there are others who'll consider the straightforwardness of the non-inline version as clean and elegant.

We don't support inline ifs to avoid bugs, which happen much easier this ways. While you or I might not make any, considering how this company is growing and how many people work on the codebase, we need some structure to maintain sanity :).

Reply:
> We don't support inline ifs to avoid bugs, which happen much easier this ways. 
Yeah, I can get behind, hence why I won't propose changing this, I can withstand spending 2 seconds adding curly braces if we can avoid the odd bug.